app/controllers/collections_controller.rb

```diff
@@ -0,0 +1,29 @@
+class CollectionsController < ApplicationController
+  before_action :store_request_in_thread, only: [:show]
+  before_action :modify_view_path
+
+  def show
+    @file_path = "#{params[:owner]}/#{params[:name]}/#{params[:path]}"
+    respond_to do |format|
+      format.html { render @file_path }
+      format.any(:png, :jpg, :jpeg) do
+        send_file "#{Rails.root}/repos/#{@file_path}.#{request.format.to_sym}"
+      end
+      format.all { render html: 'This file format cannot be rendered' }
```

Comment:
Should response a HTTP code, likes 501

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses